src/utils/canvas.js

```diff
@@ -314,3 +250,105 @@ export const determineThermometerScalePoints = (
 
   return gradationCoordinates;
 };
+
+export const generateSeaPoints = (seaBaseWidth, seaHeight, seaIndent) => {
+  // first point of line is (0, 0), i.e. start drawing shape at (0, 0) offset from shape origin (specified in x/y props)
+  const seaPointZero = [0, 0];
+
+  // note that remaining points are offsets from origin
+  // only three points are required - the 'closed' prop in the component closes the shape
+  const seaPointOne = [seaBaseWidth, 0];
+  const seaPointTwo = [seaBaseWidth + seaIndent, seaHeight];
+  const seaPointThree = [0, seaHeight];
+
+  return [...seaPointZero, ...seaPointOne, ...seaPointTwo, ...seaPointThree];
+};
+
+// given a moleculeDistribution of the form {CO2: N, H2O: M, CH4: K}, where N, M, K are integers (counts of each molecule)
+// return a flat array of the form [CO2, H2O, CO2, CH4, ...] where the molecules are distributed randomly
+const distributeMoleculesRandomly = (moleculeDistribution) => {
+  const arrayOfMolecules = [];
+  const moleculeDistributionArray = Object.entries(moleculeDistribution);
+  // moleculeDistributionArray is of the form [['Water', 5],['Methane', 10],...]
+  moleculeDistributionArray.forEach(([moleculeName, moleculeNumber]) => {
+    let counter = moleculeNumber;
+    while (counter > 0) {
+      arrayOfMolecules.push(moleculeName);
+      counter -= 1;
+    }
+  });
```

Comment:
I think you do it using array functions. Use the following to build the array containing the right number of molecules
```
Array.from({length: moleculeNumber }, x => moleculeName)
```
Then build `arrayOfMolecules` using `Array.reduce`